vestings/csv_parser.py

```diff
@@ -1,86 +1,59 @@
 from dateutil.parser import parse
-from vesting import Vesting
+from vesting import Vesting, VestingType
 from web3 import Web3
 
 CURRENT_DIRECTORY = os.path.dirname(__file__)
 
 
-def parse_vestings_csv(
-    db: orm.Session,
-    type: str,
-    chain_id: int,
-    verbose: bool,
-    start_date: int,
-    duration: int,
-):
-    vesting_file = {
-        "user": os.path.join(CURRENT_DIRECTORY, f"assets/{chain_id}/user_airdrop.csv"),
-        "user_v2": os.path.join(
-            CURRENT_DIRECTORY, f"assets/{chain_id}/user_airdrop_v2.csv"
-        ),
-        "ecosystem": os.path.join(
-            CURRENT_DIRECTORY, f"assets/{chain_id}/ecosystem_airdrop.csv"
-        ),
-        "investor": os.path.join(
-            CURRENT_DIRECTORY, f"assets/{chain_id}/investor_vestings.csv"
-        ),
-    }.get(type)
-    if not vesting_file:
-        raise ValueError(f"Not a valid vestings type: {type}")
-
-    if not os.path.exists(vesting_file):
-        print(vesting_file, "does not exist")
-
+def read_vesting_file(
+    vesting_file: str, chain_id: int, vesting_type: VestingType
+) -> List[Vesting]:
+    vestings: List[Vesting] = []
+    airdrop_address = Web3.to_checksum_address(
+        get_airdrop_addresses(chain_id)[vesting_type]
+    )
     with open(vesting_file, mode="r") as csv_file:
         csv_reader = csv.DictReader(csv_file)
-        line_count = 0
 
         print(80 * "-")
-        print(f"Processing {type} vestings")
-        print(80 * "-")
-
-        vesting_models = []
+        print(f"Processing {vesting_type.name} vestings")
 
         for row in csv_reader:
             owner = Web3.to_checksum_address(row["owner"])
-
             duration_weeks: int
-            if duration is not None:
-                duration_weeks = duration
+            start_date_timestamp: int
+
+            if "duration" in row.keys():
+                duration_weeks = int(row["duration"])
             else:
-                if "duration" in row.keys():
-                    duration_weeks = int(row["duration"])
+                if vesting_type in (VestingType.USER_V2, VestingType.INVESTOR):
+                    duration_weeks = 208
                 else:
-                    if type == "user_v2":
-                        duration_weeks = 208
-                    else:
-                        duration_weeks = 416
+                    duration_weeks = 416
 
-            start_date_timestamp: int
-            if start_date is not None:
-                start_date_timestamp = start_date
+            if "startDate" in row.keys():
+                start_date_timestamp = int(parse(row["startDate"]).timestamp())
             else:
-                if "startDate" in row.keys():
-                    start_date_timestamp = parse(row["startDate"]).timestamp()
+                if vesting_type == VestingType.USER_V2:
+                    start_date_timestamp = int(
+                        parse("2022-09-01T10:00:00+00:00").timestamp()
```

Comment:
start date could be attribute of the VestingType.

Reply:
`VestingType` is an `Enum`, but I get what you mean, maybe for next refactor 😉